src/activities/discussions/DiscussionTopicEntity.js

```diff
@@ -166,6 +189,23 @@ export class DiscussionTopicEntity extends Entity {
 		return { action, fields };
 	}
 
+	/**
+	 * Updates the topic's rating scheme selection
+	 * @param {object} topic the topic that's being modified
+	 */
+	_formatUpdateRatePostAction(topic) {
+		const { postRatingSelection } = topic || {};
+
+		if (!postRatingSelection) {
+			return;
+		}
+
+		const action = this._entity.getActionByName(Actions.discussions.topic.updateRatingType);
+		const fields = [{ name: 'ratingType', value: postRatingSelection }];
```

Comment:
I think you should check whether action exists before returning this

Reply:
Same as above, will add a check as well.